ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/ShellSnippetsInvoker.java

```diff
@@ -0,0 +1,455 @@
+        addDiagnostic(Diagnostic.debug("Result: " + result));
+        return result == null;
+    }
+
+    /* Invocation methods */
+
+    /**
+     * Invokes a method that is in the given class.
+     * The method must be a static method accepting only one parameter, a {@link Strand}.
+     *
+     * @param classLoader Class loader to find the class.
+     * @param className   Class name with the method.
+     * @param methodName  Method name to invoke.
+     * @return The result of the invocation.
+     * @throws InvokerException If invocation failed.
+     */
+    protected Object invokeScheduledMethod(ClassLoader classLoader, String className, String methodName)
+            throws InvokerException {
+        try {
+            addDiagnostic(Diagnostic.debug(String.format("Running %s.%s on schedule", className, methodName)));
+
+            // Get class and method references
+            Class<?> clazz = classLoader.loadClass(className);
+            Method method = clazz.getDeclaredMethod(methodName, Strand.class);
+            Function<Object[], Object> methodInvocation = createInvokerCallback(method);
+
+            // Schedule and run the function and return if result is valid
+            BFuture out = scheduler.schedule(new Object[1], methodInvocation, null, null, null,
+                    PredefinedTypes.TYPE_ERROR, null, null);
+            scheduler.start();
+
+            Object result = out.getResult();
+            Throwable panic = out.getPanic();
+            addDiagnostic(Diagnostic.debug("Panic: " + panic));
+            addDiagnostic(Diagnostic.debug("Result: " + result));
+
+            if (panic != null) {
+                throw new InvokerException(panic);
+            }
+            return result;
+        } catch (ClassNotFoundException e) {
+            addErrorDiagnostic(className + " class not found: " + e.getMessage());
+            throw new InvokerException(e);
+        } catch (NoSuchMethodException e) {
+            addErrorDiagnostic(methodName + " method not found: " + e.getMessage());
+            throw new InvokerException(e);
+        } catch (RuntimeException e) {
+            addErrorDiagnostic("Unexpected error: " + e.getMessage());
+            throw new InvokerException(e);
+        }
+    }
+
+    /**
+     * Invokes a method that is in the given class.
+     * This is directly invoked without scheduling.
+     * The method must be a static method accepting the given parameters.
+     *
+     * @param classLoader Class loader to find the class.
+     * @param className   Class name with the method.
+     * @param methodName  Method name to invoke.
+     * @param argTypes    Types of arguments.
+     * @param args        Arguments to provide.
+     * @return The result of the invocation.
+     * @throws InvokerException If invocation failed.
+     */
+    protected Object invokeDirectMethod(ClassLoader classLoader, String className, String methodName,
```

Comment:
Did we mean `invokeMethodDirectly`?

Reply:
The reason to name this as `invokeDirectMethod` was because its companion method name is `invokeScheduledMethod`.